Source/Scene/HereMapsImageryProvider.js

```diff
@@ -0,0 +1,521 @@
+     *
+     *
+     * @see ArcGisMapServerImageryProvider
+     * @see GoogleEarthImageryProvider
+     * @see OpenStreetMapImageryProvider
+     * @see SingleTileImageryProvider
+     * @see TileMapServiceImageryProvider
+     * @see WebMapServiceImageryProvider
+     *
+     * @see {@link https://developer.here.com/documentation/map-tile/topics/overview.html}
+     * @see {@link http://www.w3.org/TR/cors/|Cross-Origin Resource Sharing}
+     *
+     * @example
+     * var here = new Cesium.HereMapsImageryProvider({
+     *     url : 'aerial.maps.cit.api.here.com',
+     *     appId : 'get-yours-at-https://www.developer.here.com/',
+     *     appCode : 'get-yours-at-https://www.developer.here.com/',
+     * });
+     */
+    var HereMapsImageryProvider = function HereMapsImageryProvider(options) {
+        options = defaultValue(options, {});
+
+        //>>includeStart('debug', pragmas.debug);
+        if (!defined(options.appId)) {
+            throw new DeveloperError('options.appId is required.');
+        }
+        if (!defined(options.appCode)) {
+            throw new DeveloperError('options.appCode  is required.');
+        }
+        //>>includeEnd('debug');
+
+        var that = this;
+
+        this._proxy = options.proxy;
+        this._credit = new Credit({
+            text: '2014 DigitalGlobe, Inc.',
+            imageUrl: HereMapsImageryProvider._logoData,
+            link: 'http://www.here.com'
+        });
+
+        this._tilingScheme = new WebMercatorTilingScheme();
+        this._tileSize = defaultValue(options.tileSize, 256);
+
+        this._baseUrl = defaultValue(options.baseUrl, 'aerial.maps.api.here.com');
+        var mapId = 'newest';
+
+        var baseResource = new Resource({
+            url: '//{subdomain}.' + this._baseUrl,
+            queryParameters: {
+                app_id: options.appId,
+                app_code: options.appCode
+            },
+            proxy: options.proxy
+        });
+
+        var copyrightResource = baseResource.getDerivedResource({
+            url: '/maptile/2.1/copyright/' + mapId,
+            templateValues: { subdomain: '1' }
+        });
+
+        this._tileResource = baseResource.getDerivedResource({
+            url: '/maptile/2.1/{tileType}/{mapId}/{scheme}/{level}/{x}/{y}/{tileSize}/{tileFormat}',
+            templateValues: {
+                tileType: defaultValue(options.tileType, 'maptile'),
+                mapId: mapId,
+                scheme: defaultValue(options.scheme,'satellite.day'),
+                tileSize: that._tileSize,
+                tileFormat: defaultValue(options.tileFormat, 'jpg')
+            }
+        });
+
+        this._maximumLevel = 20;
+        this._urlSubdomains = ['1', '2', '3', '4'];
+
+        this._errorEvent = new CesiumEvent();
+
+        this._ready = false;
+        this._readyPromise = when.defer();
+
+        //@TODO: Do we need to support DiscardMissingTileImagePolicy?
```

Comment:
Obviously figure out if this is needed. Add a `DiscardMissingTileImagePolicy` or remove the to do. Is there a way Here tells us that there isn't a tile here?

Reply:
I'm looking into this. I can't find any mention of what they do in the case of missing tiles in their documentation, but I'll start searching Antarctica and see if I can find one 😄

Reply:
After zooming into some remote parts of the world, it appears that they always return a valid tile for any zoom level. If they don't have true imagery for it they use a rescaled version of one of the lower zoom levels. I assume this means we should just use the default DiscardMissingTileImagePolicy?